### ISSUE TITLE
AttributeError: 'MatchSpec' object has no attribute 'startswith' when installing certain packages

### ISSUE THREAD

**Issue by forrestwaters** [conda/conda#8933]
## Current Behavior

conda 4.7.7 errors out when installing molecule package

### Steps to Reproduce

```
conda install -y molecule -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: failed with current_repodata.json, will retry with next repodata source.

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/exceptions.py", line 1049, in __call__
        return func(*args, **kwargs)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/cli/main_install.py", line 20, in execute
        install(args, parser, 'install')
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/cli/install.py", line 280, in install
        force_reinstall=context.force_reinstall or context.force,
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 111, in solve_for_transaction
        force_remove, force_reinstall)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 149, in solve_for_diff
        force_remove)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 252, in solve_final_state
        ssc = self._run_sat(ssc)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 735, in _run_sat
        history_specs=ssc.specs_from_history_map)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 1194, in solve
        self.find_conflicts(specs, specs_to_add, history_specs)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 335, in find_conflicts
        bad_deps = self.build_conflict_map(specs, specs_to_add, history_specs)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 422, in build_conflict_map
        strict_channel_priority)
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 323, in _classify_bad_deps
        classes['direct'] = [_ for _ in classes['direct']
      File "/Users/fwaters/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 324, in <listcomp>
        if _[1].startswith('python ') or len(_[0]) > 1]
    AttributeError: 'MatchSpec' object has no attribute 'startswith'
```

## Expected Behavior

Package should install successfully or display information about there being a conflict

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : base
    active env location : /Users/fwaters/miniconda3
            shell level : 1
       user config file : /Users/fwaters/.condarc
 populated config files : /Users/fwaters/.condarc
          conda version : 4.7.7
    conda-build version : not installed
         python version : 3.7.3.final.0
       virtual packages :
       base environment : /Users/fwaters/miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-canary/osx-64
                          https://conda.anaconda.org/conda-canary/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/osx-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : /Users/fwaters/miniconda3/pkgs
                          /Users/fwaters/.conda/pkgs
       envs directories : /Users/fwaters/miniconda3/envs
                          /Users/fwaters/.conda/envs
               platform : osx-64
             user-agent : conda/4.7.7 requests/2.22.0 CPython/3.7.3 Darwin/18.6.0 OSX/10.14.5
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
auto_activate_base: False
channels:
  - conda-canary
  - defaults
default_channels:
  - https://repo.anaconda.com/pkgs/main
  - https://repo.anaconda.com/pkgs/r
  - https://repo.anaconda.com/pkgs/msys2
report_errors: False
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
appnope                   0.1.0                    py37_0    defaults
asn1crypto                0.24.0                   py37_0    defaults
bzip2                     1.0.7                h1de35cc_0    defaults
ca-certificates           2019.6.16            hecc5488_0    conda-forge
certifi                   2019.6.16                py37_0    defaults
cffi                      1.12.3           py37hb5b8e2f_0    defaults
chardet                   3.0.4                    py37_1    defaults
conda                     4.7.7                    py37_0    conda-canary
conda-package-handling    1.3.11                   py37_0    defaults
cryptography              2.7              py37ha12b0ac_0    defaults
decorator                 4.4.0                    py37_1    defaults
idna                      2.8                      py37_0    defaults
ipdb                      0.12                       py_0    conda-forge
ipython                   5.8.0                    py37_0    defaults
ipython_genutils          0.2.0                      py_1    conda-forge
libarchive                3.3.3                h786848e_5    defaults
libcxx                    4.0.1                hcfea43d_1    defaults
libcxxabi                 4.0.1                hcfea43d_1    defaults
libedit                   3.1.20181209         hb402a30_0    defaults
libffi                    3.2.1                h475c297_4    defaults
libiconv                  1.15                 hdd342a3_7    defaults
libxml2                   2.9.9                hf6e021a_1    defaults
lz4-c                     1.8.1.2              h1de35cc_0    defaults
lzo                       2.10                 h362108e_2    defaults
ncurses                   6.1                  h0a44026_1    defaults
openssl                   1.1.1c               h1de35cc_1    defaults
pexpect                   4.7.0                    py37_0    defaults
pickleshare               0.7.5                    py37_0    defaults
pip                       19.1.1                   py37_0    defaults
prompt_toolkit            1.0.15                   py37_0    defaults
ptyprocess                0.6.0                   py_1001    conda-forge
pycosat                   0.6.3            py37h1de35cc_0    defaults
pycparser                 2.19                     py37_0    defaults
pygments                  2.4.2                      py_0    defaults
pyopenssl                 19.0.0                   py37_0    defaults
pysocks                   1.7.0                    py37_0    defaults
python                    3.7.3                h359304d_0    defaults
python-libarchive-c       2.8                      py37_8    conda-canary
readline                  7.0                  h1de35cc_5    defaults
requests                  2.22.0                   py37_0    defaults
ripgrep                   0.10.0               hc07d326_0    defaults
ruamel_yaml               0.15.64          py37h1de35cc_0    conda-canary
setuptools                41.0.1                   py37_0    defaults
simplegeneric             0.8.1                    py37_2    defaults
six                       1.12.0                   py37_0    defaults
sqlite                    3.27.2               ha441bb4_0    defaults
tk                        8.6.8                ha441bb4_0    defaults
tqdm                      4.32.1                     py_0    defaults
traitlets                 4.3.2                    py37_0    defaults
urllib3                   1.24.2                   py37_0    defaults
wcwidth                   0.1.7                      py_1    conda-forge
wheel                     0.33.4                   py37_0    defaults
xz                        5.2.4                h1de35cc_4    defaults
yaml                      0.1.7                hc338f04_2    defaults
zlib                      1.2.11               h1de35cc_3    defaults
zstd                      1.3.7                h5bba6e5_0    defaults
```
</p></details>

**Comment by jrderuiter** [comment]
Encountering the same issue with the s3fs package.

**Comment by msarahan** [comment]
fixed by #9014
